src/core/mount.c

```diff
@@ -278,7 +278,17 @@ static int update_parameters_proc_self_mountinfo(
         if (w < 0)
                 return w;
 
-        if (r > 0 || q > 0 || w > 0)
+        /* Set flag MOUNT_PROC_JUST_CHANGED if any parameters have
+         * changed as a call to unit_add_to_dbus_queue() is needed in
+         * all cases of changed parameters.
+         *
+         * Also set flag MOUNT_PROC_JUST_CHANGED_WHAT if the "what"
+         * parameter has changed so that we know device_found_node()
+         * needs to be called.
+         */
+        if (r > 0)
+                changed = MOUNT_PROC_JUST_CHANGED_WHAT|MOUNT_PROC_JUST_CHANGED;
```

Comment:
why not just do `return MOUNT_PROC_JUST_CHANGED_WHAT|MOUNT_PROC_JUST_CHANGED;` here?